doc/manuals/Substitute.am

```diff
@@ -17,6 +17,7 @@ SUBSTITUTE=sed "\
 	s%@SYSCONFDIR@%$(sysconfdir)%g;\
 	"
 
-## Example of how to substitute:
-## squid.8: $(srcdir)/squid.8.in Makefile
-##	$(SUBSTITUTE) < $(srcdir)/squid.8.in > $@
+%.8: %.md
+	$(SUBSTITUTE) < $(srcdir)/`basename $<` | \
+	$(PANDOC) -t man -f commonmark -s -V title:`basename -s .md "$<"` -V section:8 -o $@
+
```

Comment:
```
$(PANDOC) ... -f commonmark
```

AFAICT, you have not disclosed the reasons for adopting CommonMark, so I may be missing some important caveats, but I think two competing (pandoc-supported) formats seem better suited for our purposes. They are listed here in second and third position:

1. PR-proposed [CommonMark](https://spec.commonmark.org/0.30/): Minimal subset of popular flavors. Might make switching from pandoc to something else easier.
2. [gfm](https://github.github.com/gfm/) (Github-Flavored Markdown): A strict superset of CommonMark. May be usable for non-manual pages as well (e.g., future squid.conf directive documentation), allowing us to use a single Markdown flavor for _all_ documentation written in Markdown. Supports tables and possibly footnotes (as well as ~strikethough~ text and task lists).
3. markdown (Pandoc's Markdown): This is what pandoc itself is using by default, so it is very powerful. Supports [metadata blocks](https://pandoc.org/MANUAL.html#metadata-blocks), tables, and footnotes (as well as ~strikethough~ text, task lists, and a lot more).
4. commonmark_x (CommonMark with many pandoc extensions): I do not think this is a viable candidate today due to this [[discussion](https://github.com/jgm/pandoc/issues/8391)]. I am listing it here because it sounds attractive, and I thought it might be viable before doing this research.

I am currently torn between gfm and Pandoc's Markdown. Perhaps we should use the following decision-making logic:

* If we can convince ourselves that we can (eventually) use gfm (Github-Flavored Markdown) for all Squid documentation written in Markdown, then we should use gfm because our GitHub-hosted web site (a.k.a. wiki) uses that or a superset of that format. Being able to use snippets from this repository for web site assembly can be very valuable!
* Otherwise, we should use Pandoc's Markdown for producing well-formatted manual pages (and gfm for everything else).

Currently, the biggest missing/unknown piece of information for me here is whether gfm supports metadata blocks. For example, Pandoc can take manual page title from the document metadata section. Perhaps we can even use the same metadata format for squid.conf directive documentation metadata (i.e. current `TYPE:`, `LOC:`, etc. lines). If that metadata works with gfm (without making the rendered page uglier), then perhaps gfm is the best option for us.

Reply:
My reasons for selecting `commonmark` were:
 * @rousskov insisted on being formal right now without having any idea of the amount of complexity our full set of documentation will require.
 * it is the portable subset of AFAICT **all** commonly used Markdown syntaxes. Allowing easy future changes of syntax if necessary.

Reply:
Thank you for documenting your reasons for selecting CommonMark, addressing [another change request](https://github.com/squid-cache/squid/pull/1586#discussion_r1397547985). I have checked that this change request is not materially affected by the addition of that documentation.

Reply:
I do not see any request for changes in your opening comment. Just a series of opinion statements about various alternatives and possibilities for future infrastructure.

If you do not want to demand a specific change to the format right now. Then I would like to defer that change to future PR and close off this initial upgrade.

Reply:
> I do not see any request for changes in your opening comment.

I am against the format proposed in the current PR version -- commonmark. I have suggested two alternatives (and documented why they are better). I do not yet know which one is the best for moving forward (and there may be more candidates or even different formats for different kinds of files).

> If you do not want to demand a specific change to the format right now. Then I would like to defer that change to future PR and close off this initial upgrade.

I would not describe my change request as "a demand for a specific change to the format". This PR (not me!) proposes a specific change to the existing format (of certain files). I support changing the exiting format (at least for some of the files), but not to the proposed one.

Reply:
Then please state clearly and explicitly which Markdown format you are requiring everyone to use before I go and re-write the whole PR for a fourth time to one which you will only deem inadequate.

Reply:
My proposal: replace commonmark with gfm in the pandoc invocation and leave everything else as-is.

Reply:
> Then please state clearly and explicitly which Markdown format you are requiring everyone to use

I cannot require any format in this context. As I [said](https://github.com/squid-cache/squid/pull/1586/files#r1397655257), I see two good format candidates and a usable decision making algorithm, but I do not have enough information to finalize that decision. IMHO, it is wrong to put the collection of missing information on me, but I will find the time to do it.